### DISCUSSION TITLE
RTL language retun error

### DISCUSSION THREAD

**Discussion by gitusr2211** [e107inc/e107#5359] [General]
Hello,
I start preparing new lang. with RTL direction , I change the e107_languages\English\English.php
from : 
========
setlocale(LC_ALL, 'en_GB.UTF-8', 'en_GB.utf8', 'eng_eng.utf8', 'en');
define("CORE_LC", "en");
define("CORE_LC2", "gb");
// define("TEXTDIRECTION","rtl");
========

to the following code : 
========
setlocale(LC_ALL, 'ar-ye.UTF-8', 'ar-ye.utf8', 'ar.utf8', 'ar');
define("CORE_LC", "ar");
define("CORE_LC2", "ar");
define("TEXTDIRECTION","rtl");
==========

but this return  this error at main page (/index.php)  :

Fatal error: Uncaught TypeError: defined(): Argument #1 ($constant_name) must be of type string, array given in D:\xampp\htdocs\e107_2.3.3_full\e107_handlers\news_class.php:495 Stack trace: #0 D:\xampp\htdocs\e107_2.3.3_full\e107_handlers\news_class.php(495): defined(Array) #1 D:\xampp\htdocs\e107_2.3.3_full\e107_plugins\news\news_grid_menu.php(25): news->render_newsgrid(Array) #2 D:\xampp\htdocs\e107_2.3.3_full\e107_handlers\menu_class.php(730): include('D:\\xampp\\htdocs...') #3 D:\xampp\htdocs\e107_2.3.3_full\e107_handlers\menu_class.php(612): e_menu->renderMenu('news/', 'news_grid_menu', Array) #4 D:\xampp\htdocs\e107_2.3.3_full\e107_core\shortcodes\single\menu.php(34): e_menu->renderArea('2') #5 D:\xampp\htdocs\e107_2.3.3_full\e107_handlers\shortcode_handler.php(1312): menu_shortcode('2', '') #6 [internal function]: e_parse_shortcode->doCode(Array) #7 D:\xampp\htdocs\e107_2.3.3_full\e107_handlers\shortcode_handler.php(1020): preg_replace_callback('#\\{([A-Z][^\\x02...', Array, ' {MENU=2}') #8 D:\xampp\htdocs\e107_2.3.3_full\e107_handlers\e_parse_class.php(850): e_parse_shortcode->parseCodes(' {MENU=2}', true, Object(theme_shortcodes), NULL) #9 D:\xampp\htdocs\e107_2.3.3_full\e107_handlers\e107_class.php(388): e_parse->parseTemplate(' {MENU=2}', true, Object(theme_shortcodes)) #10 D:\xampp\htdocs\e107_2.3.3_full\e107_core\templates\footer_default.php(85): e107::renderLayout('\n <div class=...', Array) #11 D:\xampp\htdocs\e107_2.3.3_full\index.php(191): include_once('D:\\xampp\\htdocs...') #12 {main} thrown in D:\xampp\htdocs\e107_2.3.3_full\e107_handlers\news_class.php on line 495

**Comment by tgtje** [comment]
@gitusr2211 welcome.
You do have to recreate the language system in meaning >
e107_languages\English\English.php  has to be e107_languages\Arabic\Arabic.php
and all language files should be in identical folderstructure in whole system.
So looking at all language folders (everywhere) you find foldername English, then a second foldername with Arabic.
Both contain same files but each has its own typed language  
Take a look at a pack [on](https://github.com/e107translations) https://github.com/e107translations how its build for the concept.
A very old Arabic pack can be found [here](https://sourceforge.net/projects/e107arabic/files/E107%20ArabiC/), but remember it dates 2007 and we are a long way from it thesedays.
Success
